src/features/metadata/components/EnvBuilds.tsx

```diff
@@ -41,30 +75,27 @@ export const EnvBuilds = ({
               fontWeight: 500,
               paddingBottom: "0"
             }}
+            data-testid="build-status"
           >
             Status: {""}
             <Typography component="span" sx={{ fontSize: "13px" }}>
-              {currentBuild.status_info ? (
-                <>
-                  {currentBuild.status} ({currentBuild.status_info})
-                </>
-              ) : (
-                <>{currentBuild.status}</>
-              )}
-              {(currentBuild.status === "Building" ||
+              {currentBuild.status}
+              {currentBuild.status_info && ` (${currentBuild.status_info})`}
+              {((currentBuild.status === "Building" ||
                 currentBuild.status === "Queued") && (
                 <CircularProgress
                   size={10}
                   sx={{
                     marginLeft: "8px"
                   }}
                 />
-              )}
+              )) ||
+                // If the selected build is a failed build, render the link to the build log.
+                (showLogLink && [". ", logLink])}
             </Typography>
           </StyledMetadataItem>
         </>
-      )}
-      {!currentBuild && (
+      ) : (
```

Comment:
I don't have the context to know whether an additional refactor here is appropriate, but you might. If you feel like this could be cleaner, please make an issue for it. In the meantime, let's get this merged.